cpp/CMakeSettings.json

```diff
@@ -0,0 +1,21 @@
+{
+  "configurations": [
+  {
+    "name": "x64-Debug (default)",
+    "generator": "Ninja",
+    "configurationType": "Debug",
+    "inheritEnvironments": [ "msvc_x64_x64" ],
+    "buildRoot": "${projectDir}\\out\\build\\${name}",
+    "installRoot": "${projectDir}\\out\\install\\${name}",
+    "cmakeCommandArgs": "",
+    "buildCommandArgs": "",
+    "ctestCommandArgs": "",
+    "variables": [
+        {
+          "name":"VCPKG_MANIFEST_INSTALL",
+          "value":"OFF"
+        }
+      ]
+    }
+  ]
+}
```

Comment:
For developers who are using Visual Studio 2019 and have vcpkg installed, when the `arrow/cpp` directory is opened in the Visual Studio IDE, it automatically identifies the `vcpkg.json` manifest and begins installing the dependencies listed in it into a subdirectory within the project. This is an undesirable behavior because the developer might want to use conda to install the dependencies or they might have already used vcpkg to install the dependencies outside the project. This `CMakeSettings.json` file sets the CMake variable `VCPKG_MANIFEST_INSTALL` to `OFF`, preventing Visual Studio from performing this automatic action. In [ARROW-11336](https://issues.apache.org/jira/browse/ARROW-11336), I will document this and instruct users to change the value to `ON` if they do wish for vcpkg to automatically install dependencies.